mysqld_exporter.go

```diff
@@ -937,35 +937,38 @@ func scrapeSlaveStatus(db *sql.DB, ch chan<- prometheus.Metric) error {
 	}
 	defer slaveStatusRows.Close()
 
-	if slaveStatusRows.Next() {
-		// There is either no row in SHOW SLAVE STATUS (if this is not a
-		// slave server), or exactly one. In case of multi-source
-		// replication, things work very much differently. This code
-		// cannot deal with that case.
-		slaveCols, err := slaveStatusRows.Columns()
-		if err != nil {
-			return err
-		}
+	// Don't bother if there is no slave status row
+	if !slaveStatusRows.Next() {
+		return nil
```

Comment:
Are you sure about that?  I know for a fact that it changes from version to version, hence the dynamic nature of this exporter.